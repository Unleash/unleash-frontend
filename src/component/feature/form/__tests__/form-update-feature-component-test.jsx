import React from 'react';
import UpdateFeatureComponent from './../form-update-feature-component';
import { shallow } from 'enzyme/build/index';

jest.mock('react-mdl');
jest.mock('../strategies-section-container', () => 'StrategiesSection');

it('render the create feature page', () => {
    let input = {
        name: 'feature',
        errors: {},
        description: 'Description',
        enabled: false,
    };
    const tree = shallow(
        <UpdateFeatureComponent
            featureToggle={input}
            onSubmit={jest.fn()}
            setValue={jest.fn()}
            addStrategy={jest.fn()}
            removeStrategy={jest.fn()}
            moveStrategy={jest.fn()}
            onCancel={jest.fn()}
            updateStrategy={jest.fn()}
            validateName={jest.fn()}
            initCallRequired={false}
            title="title"
        />
    );
    expect(tree).toMatchSnapshot();
});
