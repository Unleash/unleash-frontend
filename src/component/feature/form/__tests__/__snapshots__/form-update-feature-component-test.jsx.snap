// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the create feature page 1`] = `
<form>
  <section
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <StrategiesSection
      addStrategy={[MockFunction]}
      configuredStrategies={Array []}
      featureToggleName="feature"
      moveStrategy={[MockFunction]}
      removeStrategy={[MockFunction]}
      updateStrategy={[MockFunction]}
    />
    <br />
    <FormButtons
      onCancel={[MockFunction]}
      submitText="Update"
    />
  </section>
</form>
`;
